test/SymbolGraph/ClangImporter/SwiftAttrExtension.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %empty-directory(%t)
```

Comment:
What's the `SwiftAttrExtension` file for? My preference here is to use `split-file`, but I might be in the minority there 😅. It just makes looking at the test *so* much easier since it would all be in one place.

Reply:
I'll be honest; i have no idea 😅 all the framework skeletons i've written for the SymbolGraphGen ClangImporter tests have been copy/pasted from other framework skeletons in the test suite, and i've just not bothered to learn what each piece is needed or how to simplify it.

Reply:
For testing this case I wouldn't bother with the second `target-swift-symbolgraph-extract`. In which case all you need is the `.framework` folder with `Headers/SwiftAttrExtension.h` and `module.map`.

Though without doing `target-swift-symbolgraph-extract` a framework shouldn't be needed at all? Something like the following should work (not tested at all):
```
// REQUIRES: objc_interop

// RUN: %empty-directory(%t)
// RUN: %empty-directory(%t/SwiftAttrExtension)
// RUN: split-file %s %t

// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -enable-objc-interop -I %t/SwiftAttrExtension -module-name SwiftAttrExtension -emit-symbol-graph -emit-symbol-graph-dir %t %t/Extension.swift
// RUN: %FileCheck %s --input-file %t/SwiftAttrExtension.symbols.json

//--- SwiftAttrExtension/module.modulemap
module SwiftAttrExtension {
  header "SwiftAttrExtension.h"
}

//--- SwiftAttrExtension/SwiftAttrExtension.h
@import Foundation;

__attribute__((swift_attr("@_nonSendable(_assumed)")))
@interface MyObjcClass : NSObject
@end

//--- Extension.swift
import SwiftAttrExtension

extension MyObjcClass {
  // CHECK: s:So11MyObjcClassC18SwiftAttrExtensionE8someFuncyyF
  public func someFunc() {}
}
```

Reply:
I had to add `-emit-module` and `-import-underlying-module` to get the test to work on my machine, but that's a lot simpler, thanks so much!

Reply:
Did you add the `import SwiftAttrExtension`? Those would be required without it, since otherwise there's nothing importing that module. `-import-underlying-module` with `-emit-module` works since now the ObjC module `SwiftAttrExtension` is the "underlying module".

Reply:
With `-emit-module` and the `import` line but without `-import-underlying-module`, i got the following errors:

```
/Users/victoria_m/swift-base/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/SymbolGraph/ClangImporter/Output/SwiftAttrExtension.swift.tmp/Extension.swift:1:8: warning: file 'Extension.swift' is part of module 'SwiftAttrExtension'; ignoring import
import SwiftAttrExtension
       ^
/Users/victoria_m/swift-base/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/SymbolGraph/ClangImporter/Output/SwiftAttrExtension.swift.tmp/Extension.swift:3:11: error: cannot find type 'MyObjcClass' in scope
extension MyObjcClass {
          ^~~~~~~~~~~
```

After taking out the import line and adding `-import-underlying-module`, the test passed.

Reply:
Oh, that's just because the module name is the same. Should just be able to make it something else.

Reply:
That changes the nature of the test, though. The extension symbol being checked is no longer in the "same" module as the class being extended, which was part of the original reproduction. I would rather keep the class and its extension in the same module instead of splitting them apart.

Reply:
Does that actually matter here? IIUC the issue should be for *any* imported ObjC code, not just the underlying module.